src/network/netdev/l2tp-tunnel.c

```diff
@@ -0,0 +1,500 @@
+                                  const char *filename,
+                                  unsigned line,
+                                  const char *section,
+                                  unsigned section_line,
+                                  const char *lvalue,
+                                  int ltype,
+                                  const char *rvalue,
+                                  void *data,
+                                  void *userdata) {
+        L2tpTunnel *t = userdata;
+        uint16_t port;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = parse_ip_port(rvalue, &port);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse L2TP Tunnel's '%s' port '%s'.", lvalue, rvalue);
+                return 0;
+        }
+
+        if (streq(lvalue, "UDPSourcePort"))
+                t->l2tp_udp_sport = port;
+        else
+                t->l2tp_udp_dport = port;
+
+        return 0;
+}
+
+static int netdev_l2tp_tunnel_verify(NetDev *netdev, const char *filename) {
+        L2tpTunnel *t;
+
+        assert(netdev);
+        assert(filename);
+
+        t = L2TP(netdev);
+
+        assert(t);
+
+        if (!IN_SET(t->family, AF_INET, AF_INET6)) {
+                log_netdev_error(netdev,
+                                 "L2TP Tunnel with invalid address family configured in %s. Ignoring", filename);
+                return -EINVAL;
+        }
+
+        if (in_addr_is_null(t->family, &t->local) || in_addr_is_null(t->family, &t->remote)) {
+                log_netdev_error(netdev,
+                                 "L2TP Tunnel without a local or remote address configured in %s. Ignoring", filename);
+                return -EINVAL;
+        }
+
+        return 0;
+}
+
+int config_parse_l2tp_session_id(const char *unit,
+                                 const char *filename,
+                                 unsigned line,
+                                 const char *section,
+                                 unsigned section_line,
+                                 const char *lvalue,
+                                 int ltype,
+                                 const char *rvalue,
+                                 void *data,
+                                 void *userdata) {
+        _cleanup_free_ L2tpSession *session = NULL;
+        L2tpTunnel *t = userdata;
+        uint32_t k;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = l2tp_tunnel_new_session(t, section_line, &session);
+        if (r < 0)
+                return r;
+
+        r = safe_atou32(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, 0, "Failed to parse L2TP tunnel session id. Ignoring assignment: %s", rvalue);
+                return 0;
+        }
```

Comment:
if i read the man page you added correctly, "0" is not a valid value here? should be refused?